runtime-sdk/src/dispatcher.rs

```diff
@@ -250,6 +250,10 @@ impl<R: Runtime> Dispatcher<R> {
             }
         }
 
+        if let Err(e) = R::Modules::before_authorized_call_dispatch(ctx, &call) {
```

Comment:
It looks like we have 2 method authorizations, one here and one few lines above (`method_authorizer`). A bit confusing. Can this be merged?

Reply:
I don't think so. One is for caller-specified dispatch options, the other is a hook for that runtime modules can implement. So they serve two different purposes and we need both.